docs/user-guide/data-structures/creating-arrays-and-datasets.ipynb

```diff
@@ -0,0 +1,527 @@
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "sc.arange('x', start=-2, stop=5, step=1.2, unit='K')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`arange` does not include the stop value but `linspace` does by default.\n",
+    "Please note that the caveats described in [NumPy's documentation](https://numpy.org/doc/stable/user/basics.creation.html#d-array-creation-functions) apply to Scipp as well.\n",
+    "\n",
+    "#### Filling with a Value\n",
+    "\n",
+    "There are a number opf functions to create N-D arrays with a fixed value, e.g. [scipp.zeros](../../generated/functions/scipp.zeros.rst) and [scipp.full](../../generated/functions/scipp.full.rst).\n",
+    "`scipp.zeros` creates a variable of any number of dimensions filled with zeros:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "sc.zeros(dims=['x', 'y'], shape=[3, 4])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "And `scipp.full` creates a variable filled with a given value:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sc.full(dims=['x', 'y'], shape=[3, 4], value=1.23)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Special DTypes\n",
+    "\n",
+    "Scipp has a number of `dtypes` that require some conversion when creating variables.\n",
+    "Notably [scipp.datetimes](../../generated/functions/scipp.datetimes.rst), [scipp.vectors](../../generated/functions/scipp.vectors.rst), and their scalar counterparts [scipp.datetime](../../generated/functions/scipp.datetime.rst), [scipp.vector](../../generated/functions/scipp.vector.rst).\n",
+    "As well as types for spatial transformations in [scipp.spatial](../../generated/modules/scipp.spatial.rst).\n",
+    "While variables of all of these dtypes can be constructed using `scipp.array` and `scipp.scalar`, the specialized functions offer more convenience and document their intent better.\n",
+    "\n",
+    "`scipp.datetimes` constructs an array of date-time-points.\n",
+    "It can be called either with strings in ISO 8601 format:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "sc.datetimes(dims=['t'], values=['2021-01-10T01:23:45',\n",
+    "                                 '2021-01-11T23:45:01'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Or with integers which encode the number of time units elapsed since the Unix epoc.\n",
+    "See also [scipp.epoch](../../generated/functions/scipp.epoch.rst)."
```

Comment:
Since (iirc) `arange` and `linspace` to not support datetimes, it might be useful to add an example of how to achieve this, .e.g,:
```python
sc.epoch(unit='s') + sc.arange('time', 10, unit='s')
```

Reply:
`arange` does. Not sure we need to mention this explicitly.

Reply:
I think we *should* mention how to create datetimes with `arange`. It is not obvious to me.